lib/ansible/modules/monitoring/zabbix/zabbix_proxy.py

```diff
@@ -89,14 +98,44 @@
         required: false
         choices: ['present', 'absent']
         default: "present"
+        type: str
     interface:
         description:
             - Dictionary with params for the interface when proxy is in passive mode
-            - 'Available values are: dns, ip, main, port, type and useip.'
-            - Please review the interface documentation for more information on the supported properties
-            - U(https://www.zabbix.com/documentation/3.2/manual/api/reference/proxy/object#proxy_interface)
+            - For more information, review proxy interface documentation at
+            - U(https://www.zabbix.com/documentation/4.0/manual/api/reference/proxy/object#proxy_interface)
         required: false
+        suboptions:
```

Comment:
Probably it makes sense to also add `options=dict(...)` to `interface` in `argument_spec`. Since that can break existing playbooks/roles, it's probably better to do that in a follow-up PR (which should not be backported).

Reply:
Thanks, wanted to add it within this patch, but I've thought the same. Will open a new PR for this.